---
title: "nf-core/mag"
author: "Alex Caswell"
date: "2025-05-14"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r imports}

library(plyr)
library(tidyverse)
library(ggtext)
library(cowplot)

```

``` {r get_dastool_summaries, results = 'hide'}

dastool_summaries <- data.frame()

for (filename in list.files("data/mag/GenomeBinning/DASTool")) {
  if (length(dastool_summaries) == 0) {
    dastool_summaries <- read_tsv(paste0("data/mag/GenomeBinning/DASTool/", filename), col_names = TRUE)
  } else {
    dastool_summaries <- rbind(dastool_summaries, read_tsv(paste0("data/mag/GenomeBinning/DASTool/", filename), col_names = TRUE))
  }
}

names(dastool_summaries)[names(dastool_summaries) == 'bin'] <- 'user_genome'

```

```{r get_gtdb}

dastool_bac120 <- read_tsv("data/gtdbtk/nfcoremag/gtdbtk.bac120.summary.tsv")
dastool_bac120_extra <- read_tsv("data/gtdbtk/nfcoremag/r012_r018.gtdbtk.bac120.summary.tsv")
dastool_ar53 <- read_tsv("data/gtdbtk/nfcoremag/gtdbtk.ar53.summary.tsv")

dastool_gtdb <- rbind(
    dastool_bac120,
    dastool_bac120_extra,
    dastool_ar53
  ) %>% 
  filter(!grepl("unbinned", user_genome)) %>%
  mutate(SampleName=str_extract(user_genome, "[R]+[0-9]+")) %>%
  mutate(Bin=str_extract(user_genome, "(?<=\\.)[0-9]+")) %>%
  mutate(Assembler=str_extract(user_genome, "FLYE|METAMDBG")) %>%
  mutate(Binner=str_extract(user_genome, "MetaBAT2|MaxBin2")) %>%
  mutate(Kingdom = str_replace_all(sapply(strsplit(classification, ";", fixed=TRUE), '[', 1), c("d__"=""))) %>% relocate(`Kingdom`, .before = c(1)) %>%
  mutate(Phylum = str_replace_all(sapply(strsplit(classification, ";", fixed=TRUE), '[', 2), c("a_A"="a", "a_B"="a", "a_C"="a", "_I"="", "p__"=""))) %>% relocate(`Phylum`, .before = c(2)) %>%
  mutate(Class = str_replace_all(sapply(strsplit(classification, ";", fixed=TRUE), '[', 3), c("c__"=""))) %>% relocate(`Class`, .before = c(3)) %>%
  mutate(Order = str_replace_all(sapply(strsplit(classification, ";", fixed=TRUE), '[', 4), c("o__"=""))) %>% relocate(`Order`, .before = c(4)) %>%
  mutate(Family = str_replace_all(sapply(strsplit(classification, ";", fixed=TRUE), '[', 5), c("f__"=""))) %>% relocate(`Family`, .before = c(5)) %>%
  mutate(Genus = str_replace_all(sapply(strsplit(classification, ";", fixed=TRUE), '[', 6), c("g__"=""))) %>% relocate(`Genus`, .before = c(6)) %>%
  mutate(Species = str_replace_all(sapply(strsplit(classification, ";", fixed=TRUE), '[', 7), c("s__"=""))) %>% relocate(`Species`, .before = c(7)) %>%
  left_join(dastool_summaries, by = "user_genome") %>%
  subset(select = -c(`closest_genome_taxonomy`, 
                     `other_related_references(genome_id,species_name,radius,ANI,AF)`,
                     `closest_placement_taxonomy`,
                     `pplacer_taxonomy`,
                     `user_genome`,
                     `bin_set`,
                     `classification`))  %>% mutate(MIMAG = ifelse(SCG_completeness > 90 & SCG_redundancy < 5, "HQ", ifelse(SCG_completeness > 50 & SCG_redundancy < 10, "MQ", "LQ"))) %>%
  mutate(MIMAG = fct_relevel(MIMAG, "HQ", "MQ", "LQ"))


```


```{r visualize_mags}

visualize_mags <- ggplot(dastool_gtdb[dastool_gtdb$SCG_completeness > 95, ] %>% drop_na(), aes(x = size, y = N50, colour=SampleName)) + 
  geom_point(aes(size = size)) + 
  facet_wrap(. ~ Assembler) 

visualize_mags

```


```{r boxplot_of_completeness}

nfmag_boxplot_completeness <- ggplot(dastool_gtdb, aes(x = `SampleName`, y=SCG_completeness, fill=SampleName)) +
  geom_boxplot(alpha = 0.8, show.legend = FALSE) + 
  scale_y_continuous(breaks = seq(0, 150, 20)) + 
  expand_limits(y = c(0, 100)) +
  scale_fill_brewer(palette = "Set1") + 
  facet_wrap(. ~ Assembler) +
  xlab("") +
  ylab("Completeness (%)") +
  theme_classic() +
  theme(
    axis.text.y = element_markdown(size = 16),
    axis.text.x = element_markdown(size = 14),
    axis.title.x = element_markdown(size = 12),
    axis.title.y = element_markdown(size = 12),
    panel.spacing = unit(1, "lines"),
    strip.text = element_markdown(face = "bold", size = 12),
    panel.background = element_rect(fill = '#fff7ef00'),
    plot.background = element_rect(fill = '#fff7ef00'),
    legend.background = element_rect(fill = '#fff7ef00'),
    legend.text = element_markdown(size = 12),
    legend.title = element_blank(),
  )

ggsave("graphs/nfmag/nfmag_boxplot_completeness.png", nfmag_boxplot_completeness)
nfmag_boxplot_completeness

```

```{r histogram_of_completeness}

completeness <- ggplot(dastool_gtdb, aes(x = `SCG_completeness`, fill=SampleName)) +
  geom_histogram(binwidth = 10, boundary = 0, closed = "left", col="black") + 
  scale_x_continuous(breaks = seq(0, 100, 10)) +
  scale_y_continuous(breaks = seq(0, 850, 50)) + 
  expand_limits(x = c(0, 100)) +
  scale_fill_brewer(palette = "Set1") + 
  coord_flip() +
  guides(fill = guide_legend(ncol = 6)) +
  ylab("\\# of MAGs") +
  xlab("Completeness (%)") +
  theme_classic() +
  facet_wrap(. ~ Assembler) +
  theme(
    axis.text.y = element_markdown(size = 10),
    axis.text.x = element_markdown(size = 16),
    axis.title.x = element_markdown(size = 12),
    axis.title.y = element_markdown(size = 12),
    panel.spacing = unit(1, "lines"),
    strip.text = element_markdown(face = "bold", size = 12),
    panel.background = element_rect(fill = '#fff7ef00'),
    plot.background = element_rect(fill = '#fff7ef00'),
    legend.background = element_rect(fill = '#fff7ef00'),
    legend.text = element_markdown(size = 12),
    legend.title = element_blank(),
    legend.position = "bottom",
  )

completeness
ggsave("graphs/nfmag/completeness_histogram.png", completeness)

```


```{r number_of_bins}

#dastool_count_bins <- data.frame(
#  SampleName = names(table(dastool_gtdb$SampleName)),
#  bin_count = as.vector(table(dastool_gtdb$SampleName))
#)

dastool_count_bins <- dastool_gtdb %>%
  dplyr::count(SampleName, Assembler)

nfmag_number_of_bins <- ggplot(dastool_count_bins, aes(x = SampleName, y = n)) + 
  geom_bar(stat="identity", colour = "black", show.legend = FALSE, fill="grey") +
  geom_text(label = dastool_count_bins$n, size = 6, vjust = -0.2) +
 # scale_fill_brewer(palette = "Set1") +
  scale_x_discrete(labels = c("R003", "R004", "R005", "R007", "R012", "R018")) +
  ylab("Total \\# of Genome Bins") +
  xlab("") +
  expand_limits(y = c(0, 300)) +
  theme_classic() +
  facet_wrap(. ~ Assembler) + 
  theme(
    axis.text.y = element_markdown(size = 16),
    axis.text.x = element_markdown(size = 14),
    axis.title.x = element_markdown(size = 10),
    axis.title.y = element_markdown(size = 10),
    panel.spacing = unit(1, "lines"),
    strip.text = element_markdown(face = "bold", size = 12),
    panel.background = element_rect(fill = '#fff7ef00'),
    plot.background = element_rect(fill = '#fff7ef00'),
    legend.background = element_rect(fill = '#fff7ef00'),
    legend.text = element_markdown(size = 12),
    legend.title = element_blank()
  )

nfmag_number_of_bins
ggsave("graphs/nfmag/number_of_bins.png", nfmag_number_of_bins)

```

```{r bin_n50}


#dastool_avg_bins_n50 <- aggregate(dastool_gtdb, by = list(dastool_gtdb$SampleName), FUN = "mean")[, c("Group.1", "N50")] %>%
#  setNames(c("SampleName", "N50"))

dastool_avg_bins_n50 <- ddply(dastool_gtdb, c("SampleName", "Assembler"), summarise, N50 = mean(N50))

bin_n50 <- ggplot(dastool_avg_bins_n50, aes(x = SampleName, y = round(N50/1000), fill = SampleName)) + 
  geom_bar(stat="identity", show.legend = FALSE, colour="black", fill="grey") +
  geom_text(label = round(dastool_avg_bins_n50$N50/1000), size = 6, vjust = -0.2) +
  scale_fill_brewer(palette = "Set1") +
  scale_x_discrete(labels = c("R003", "R004", "R005", "R007", "R012", "R018")) +
  scale_y_continuous(labels = (format_si())) +
  ylab("Average Bin N50 (Kbp)") +
  xlab("") +
  expand_limits(y = c(0, 2200)) +
  theme_classic() +
  facet_wrap(. ~ Assembler) +
  theme(
    axis.text.y = element_markdown(size = 16),
    axis.text.x = element_markdown(size = 14),
    axis.title.x = element_markdown(size = 10),
    axis.title.y = element_markdown(size = 10),
    panel.spacing = unit(1, "lines"),
    strip.text = element_markdown(face = "bold", size = 12),
    panel.background = element_rect(fill = '#fff7ef00'),
    plot.background = element_rect(fill = '#fff7ef00'),
    legend.background = element_rect(fill = '#fff7ef00'),
    legend.text = element_markdown(size = 12),
    legend.title = element_blank()
  )

bin_n50
ggsave("graphs/nfmag/bin_n50.png", bin_n50)

```

```{r make_numbers_and_n50_panel}

numbers_and_n50_panel <- plot_grid(nfmag_number_of_bins, bin_n50, labels = c("A", "B"), ncol = 1)

ggsave("graphs/nfmag/numbers_and_n50_panel.png", numbers_and_n50_panel, units = "px", width=2800, height=1600)
numbers_and_n50_panel

```

```{r make_completeness_panel}

nfmag_completeness_panel <- plot_grid(nfmag_boxplot_completeness, completeness, labels = c("A", "B"), ncol = 1, rel_widths = c(1, 1.2))

ggsave("graphs/nfmag/nfmag_completeness_panel.png", nfmag_completeness_panel, units = "px", width = 2800, height = 1800)
nfmag_completeness_panel

```

```{r join_with_kmer_abundance}

dastool_gtdb_abundances <- inner_join(
  abundances_with_taxonomy,
  dastool_gtdb %>% rename_at('Species',),
  by="species"
)

```


```{r barchart_mimags}

all_bins_mimags <- rbind(
  dastool_gtdb[dastool_gtdb$Assembler == "METAMDBG", c("SampleName", "Bin", "MIMAG")] %>% add_column(Pipeline = c("nf-core/mag (METAMDBG)")),
  dastool_gtdb[dastool_gtdb$Assembler == "FLYE", c("SampleName", "Bin", "MIMAG")] %>% add_column(Pipeline = c("nf-core/mag (FLYE)")),
  busco_all[busco_all$Complete > 50, c("SampleName", "Bin", "MIMAG")] %>% add_column(Pipeline = "LOMA")
)

mimag_barchart <- ggplot(all_bins_mimags, aes(x = SampleName, fill = MIMAG)) + 
  geom_bar() + 
  facet_wrap(. ~ Pipeline) +
  xlab("") + 
  ylab("Number of Genomic Bins (>50% Complete)") +
  guides(fill = guide_legend("MIMAG Classification:", ncol = 3)) +
  scale_fill_discrete(labels = c("High-Quality Draft", "Medium-Quality Draft", "Low-Quality Draft")) +
  theme_classic() +
  theme(
    axis.text.y = element_markdown(size = 16),
    axis.text.x = element_markdown(size = 10),
    axis.title.x = element_markdown(size = 10),
    axis.title.y = element_markdown(size = 10),
    panel.spacing = unit(1, "lines"),
    strip.text = element_markdown(face = "bold", size = 12),
    panel.background = element_rect(fill = '#fff7ef00'),
    plot.background = element_rect(fill = '#fff7ef00'),
    legend.background = element_rect(fill = '#fff7ef00'),
    legend.text = element_markdown(size = 12),
    legend.position = "bottom"
   # legend.title = element_blank()
  )

ggsave("graphs/nfmag/mimag_barchart.png", mimag_barchart, units = "px", width = 2800, height = 1200)
mimag_barchart


```

```{r get_all_gtdb_bins}

all_gtdb_bins <- rbind(
  dastool_gtdb[, c("SampleName", "Assembler", "Phylum")],
  busco_gtdb[, c("SampleName", "Phylum")] %>% add_column(Assembler = "LOMA")
)

```

```{r phylum}



mags_compare_phyla <- ggplot(all_gtdb_bins %>% dplyr::count(SampleName, Assembler, Phylum), aes(x=SampleName, y = n, fill=Phylum)) + 
  geom_bar(stat="identity", na.rm=TRUE, position = "fill", show.legend=TRUE) + 
  facet_wrap(~ Assembler) + 
  coord_flip() +
  xlab("") +
  ylab("") +
  #ggtitle("<b>Distribution of Phyla</b> (k-mer classification)") +
  labs(fill = "Phylum") +
  scale_y_continuous(labels = scales::percent, breaks=c(0, 1)) +
  #scale_fill_manual(values = colorRampPalette(brewer.pal(8, "Set3"))(length(unique((dastool_gtdb %>% dplyr::count(SampleName, Assembler, Phylum)$Phylum))))) +
  theme_classic() +
 # guides(y = "none") +
  theme(
    axis.text.y = element_markdown(face="bold", size = 14),
    axis.text.x = element_markdown(size = 12),
    plot.title = element_markdown(hjust = 0.5),
    panel.spacing = unit(1, "lines"),
    strip.text = element_markdown(face = "bold", size = 12),
    legend.margin = margin(l = -20, t = -20),
    legend.text = element_markdown(size = 12),
    legend.title = element_blank(),
    legend.position = "bottom",
  )

ggsave("graphs/nfmag/mags_compare_phyla.png", mags_compare_phyla, width = 2187, height = 900, units="px")
mags_compare_phyla


```

```{r get_coverm}

coverm_summaries <- data.frame()

for (filename in list.files("data/coverm/FLYE")) {
  if (length(coverm_summaries) == 0) {
    coverm_summaries <- read_tsv(paste0("data/coverm/FLYE/", filename), col_names = TRUE)
    colnames(coverm_summaries) <- c("Genome", "Relative Abundance", "Mean", "Trimmed Mean", "Covered Bases", "Variance")
    coverm_summaries$Genome[which(coverm_summaries$Genome == "unmapped")] <- paste0("unmapped_", filename)
  } else {
    summary <- read_tsv(paste0("data/coverm/FLYE/", filename), col_names = TRUE)
    colnames(summary) <- c("Genome", "Relative Abundance", "Mean", "Trimmed Mean", "Covered Bases", "Variance")
    summary$Genome[which(summary$Genome == "unmapped")] <- paste0("unmapped_", filename)
    coverm_summaries <- rbind(coverm_summaries, summary)
  }
}

for (filename in list.files("data/coverm/METAMDBG")) {
  if (length(coverm_summaries) == 0) {
    coverm_summaries <- read_tsv(paste0("data/coverm/METAMDBG/", filename), col_names = TRUE)
    colnames(coverm_summaries) <- c("Genome", "Relative Abundance", "Mean", "Trimmed Mean", "Covered Bases", "Variance")
    coverm_summaries$Genome[which(coverm_summaries$Genome == "unmapped")] <- paste0("unmapped_", filename)
  } else {
    summary <- read_tsv(paste0("data/coverm/METAMDBG/", filename), col_names = TRUE)
    colnames(summary) <- c("Genome", "Relative Abundance", "Mean", "Trimmed Mean", "Covered Bases", "Variance")
    summary$Genome[which(summary$Genome == "unmapped")] <- paste0("unmapped_", filename)
    coverm_summaries <- rbind(coverm_summaries, summary)
  }
}

coverm_summaries <- coverm_summaries %>%
  mutate(SampleName=str_extract(Genome, "[R]+[0-9]+")) %>%
  mutate(Bin=str_extract(Genome, "(?<=\\.)[0-9]+")) %>%
  mutate(Assembler=str_extract(Genome, "FLYE|METAMDBG")) %>%
  mutate(Binner=str_extract(Genome, "MetaBAT2|MaxBin2")) %>%
  subset(select = -c(`Genome`))

coverm_gtdb <- left_join(coverm_summaries, dastool_gtdb, by = c("SampleName", "Bin", "Assembler", "Binner"))

```

```{r domain_compare}

names(coverm_gtdb)[names(coverm_gtdb) == "Kingdom"] <- "Domain"

kmer_domain_proportions <- domain_proportions 
colnames(kmer_domain_proportions) <- c("Domain", "Assembler", "SampleName", "Relative Abundance") 
kmer_domain_proportions <- kmer_domain_proportions %>%
  add_column(Approach = c("K-mers"))

all_domain_proportions <- coverm_gtdb[, c("Domain", "Assembler", "Relative Abundance", "SampleName")] %>% 
  aggregate(. ~ Domain + Assembler + SampleName, FUN = sum) %>% 
  add_column(Approach = c("MAGs")) %>% 
  rbind(kmer_domain_proportions)

coverm_domains <- ggplot(all_domain_proportions, aes(x = Assembler, y = `Relative Abundance`, fill = `Domain`)) + 
  geom_bar(stat = "identity", position = "fill") +
  facet_wrap(~ SampleName) + 
  coord_flip() +
  xlab("") +
  ylab("") +
  #ggtitle("<b>MAG and k-mer-based taxonomy, by Phylum</b>") +
  labs(fill = "Domain") +
  scale_y_continuous(labels = scales::percent, breaks=c(0, 1)) +
  theme_classic() +
  theme(
    axis.text.y = element_markdown(size = 10),
    axis.text.x = element_markdown(size = 12),
    plot.title = element_markdown(hjust = 0.5),
    panel.spacing = unit(1, "lines"),
    strip.text = element_markdown(face = "bold", size = 12),
    legend.margin = margin(l = -20, t = -20),
    legend.text = element_markdown(size = 12),
    legend.title = element_blank(),
    legend.position = "bottom",
  )

coverm_domains
htmlwidgets::saveWidget(as_widget(plotly_build(coverm_domains)), "graphs/plotly/domain.html")

```

```{r phylum_compare}

kmer_phyla_proportions <- phyla_proportions
colnames(kmer_phyla_proportions) <- c("Phylum", "Assembler", "SampleName", "Relative Abundance") 
kmer_phyla_proportions <- kmer_phyla_proportions %>%
  filter(Assembler != "MetaPhlAn (Prodigal)") %>%
  filter(Assembler != "MetaPhlAn (1Kbp Shreds)") %>%
  add_column(Approach = c("K-mers"))

all_phyla_proportions <- coverm_gtdb[, c("Phylum", "Assembler", "Relative Abundance", "SampleName")] %>% 
  aggregate(. ~ Phylum + Assembler + SampleName, FUN = sum) %>% 
  add_column(Approach = c("MAGs")) %>% 
  rbind(kmer_phyla_proportions)

coverm_phyla <- ggplot(all_phyla_proportions, aes(x = Assembler, y = `Relative Abundance`, fill = `Phylum`)) + 
  geom_bar(stat = "identity", position = "fill") +
  facet_wrap(~ SampleName) + 
  coord_flip() +
  xlab("") +
  ylab("") +
  #ggtitle("<b>MAG and k-mer-based taxonomy, by Phylum</b>") +
  labs(fill = "Phylum") +
  scale_y_continuous(labels = scales::percent, breaks=c(0, 1)) +
  theme_classic() +
  theme(
    axis.text.y = element_markdown(size = 10),
    axis.text.x = element_markdown(size = 12),
    plot.title = element_markdown(hjust = 0.5),
    panel.spacing = unit(1, "lines"),
    strip.text = element_markdown(face = "bold", size = 12),
    legend.margin = margin(l = -20, t = -20),
    legend.text = element_markdown(size = 12),
    legend.title = element_blank(),
    legend.position = "bottom",
  )

ggsave("graphs/nfmag/compare_phyla.png", coverm_phyla, units = "px", width = 2100, height = 1400)
coverm_phyla
htmlwidgets::saveWidget(as_widget(plotly_build(coverm_phyla)), "graphs/plotly/phylum.html")

```

```{r class_compare}

kmer_class_proportions <- classes_proportions 
colnames(kmer_class_proportions) <- c("Class", "Assembler", "SampleName", "Relative Abundance") 
kmer_class_proportions <- kmer_class_proportions %>%
  add_column(Approach = c("K-mers"))

all_class_proportions <- coverm_gtdb[, c("Class", "Assembler", "Relative Abundance", "SampleName")] %>% 
  aggregate(. ~ Class + Assembler + SampleName, FUN = sum) %>% 
  add_column(Approach = c("MAGs")) %>% 
  rbind(kmer_class_proportions)

coverm_class <- ggplot(all_class_proportions, aes(x = Assembler, y = `Relative Abundance`, fill = `Class`)) + 
  geom_bar(stat = "identity", position = "fill") +
  facet_wrap(~ SampleName) + 
  coord_flip() +
  xlab("") +
  ylab("") +
  #ggtitle("<b>Relative Abundance of MAGs, by Class</b>") +
  labs(fill = "Class") +
  scale_y_continuous(labels = scales::percent, breaks=c(0, 1)) +
  theme_classic() +
  theme(
    axis.text.y = element_markdown(face="bold", size = 14),
    axis.text.x = element_markdown(size = 12),
    plot.title = element_markdown(hjust = 0.5),
    panel.spacing = unit(1, "lines"),
    strip.text = element_markdown(face = "bold", size = 12),
    legend.margin = margin(l = -20, t = -20),
    legend.text = element_markdown(size = 12),
    legend.title = element_blank(),
    legend.position = "bottom",
  )

coverm_class
htmlwidgets::saveWidget(as_widget(plotly_build(coverm_class)), "graphs/plotly/class.html")

```

```{r order_compare}

kmer_order_proportions <- orders_proportions 
colnames(kmer_order_proportions) <- c("Order", "Assembler", "SampleName", "Relative Abundance") 
kmer_order_proportions <- kmer_order_proportions %>%
  add_column(Approach = c("K-mers"))

all_order_proportions <- coverm_gtdb[, c("Order", "Assembler", "Relative Abundance", "SampleName")] %>% 
  aggregate(. ~ Order + Assembler + SampleName, FUN = sum) %>% 
  add_column(Approach = c("MAGs")) %>% 
  rbind(kmer_order_proportions)

coverm_order <- ggplot(all_order_proportions, aes(x = Assembler, y = `Relative Abundance`, fill = `Order`)) + 
  geom_bar(stat = "identity", position = "fill") +
  facet_wrap(~ SampleName) + 
  coord_flip() +
  xlab("") +
  ylab("") +
 # ggtitle("<b>Relative Abundance of MAGs, by Order</b>") +
  labs(fill = "Order") +
  scale_y_continuous(labels = scales::percent, breaks=c(0, 1)) +
  theme_classic() +
  theme(
    axis.text.y = element_markdown(face="bold", size = 14),
    axis.text.x = element_markdown(size = 12),
    plot.title = element_markdown(hjust = 0.5),
    panel.spacing = unit(1, "lines"),
    strip.text = element_markdown(face = "bold", size = 12),
    legend.margin = margin(l = -20, t = -20),
    legend.text = element_markdown(size = 12),
    legend.title = element_blank(),
    legend.position = "bottom",
  )

coverm_order
htmlwidgets::saveWidget(as_widget(plotly_build(coverm_order)), "graphs/plotly/order.html")

```

```{r family_compare}

kmer_family_proportions <- families_proportions 
colnames(kmer_family_proportions) <- c("Family", "Assembler", "SampleName", "Relative Abundance") 
kmer_family_proportions <- kmer_family_proportions %>%
  add_column(Approach = c("K-mers"))

all_family_proportions <- coverm_gtdb[, c("Family", "Assembler", "Relative Abundance", "SampleName")] %>% 
  filter(Family != "") %>%
  aggregate(. ~ Family + Assembler + SampleName, FUN = sum) %>% 
  add_column(Approach = c("MAGs")) %>% 
  rbind(kmer_family_proportions)

coverm_family <- ggplot(all_family_proportions, aes(x = Assembler, y = `Relative Abundance`, fill = `Family`)) + 
  geom_bar(stat = "identity", position = "fill") +
  facet_wrap(~ SampleName) + 
  coord_flip() +
  xlab("") +
  ylab("") +
  #ggtitle("<b>Relative Abundance of MAGs, by Family</b>") +
  labs(fill = "Family") +
  scale_y_continuous(labels = scales::percent, breaks=c(0, 1)) +
  theme_classic() +
  theme(
    axis.text.y = element_markdown(face="bold", size = 14),
    axis.text.x = element_markdown(size = 12),
    plot.title = element_markdown(hjust = 0.5),
    panel.spacing = unit(1, "lines"),
    strip.text = element_markdown(face = "bold", size = 12),
    legend.margin = margin(l = -20, t = -20),
    legend.text = element_markdown(size = 12),
    legend.title = element_blank(),
    legend.position = "bottom",
  )

coverm_family
htmlwidgets::saveWidget(as_widget(plotly_build(coverm_family)), "graphs/plotly/family.html")

```


```{r genus_compare}

kmer_genus_proportions <- genus_proportions 
colnames(kmer_genus_proportions) <- c("Genus", "Assembler", "SampleName", "Relative Abundance") 
kmer_genus_proportions <- kmer_genus_proportions %>%
  add_column(Approach = c("K-mers"))

all_genus_proportions <- coverm_gtdb[, c("Genus", "Assembler", "Relative Abundance", "SampleName")] %>% 
  filter(Genus != "") %>%
  filter(Genus != " ") %>%
  aggregate(. ~ Genus + Assembler + SampleName, FUN = sum) %>% 
  add_column(Approach = c("MAGs")) %>% 
  rbind(kmer_genus_proportions)

coverm_genus <- ggplot(all_genus_proportions, aes(x = Assembler, y = `Relative Abundance`, fill = `Genus`)) + 
  geom_bar(stat = "identity", position = "fill") +
  facet_wrap(~ SampleName) + 
  coord_flip() +
  xlab("") +
  ylab("") +
  #ggtitle("<b>Relative Abundance of MAGs, by Genus</b>") +
  labs(fill = "Genus") +
  scale_y_continuous(labels = scales::percent, breaks=c(0, 1)) +
  theme_classic() +
  theme(
    axis.text.y = element_markdown(face="bold", size = 14),
    axis.text.x = element_markdown(size = 12),
    plot.title = element_markdown(hjust = 0.5),
    panel.spacing = unit(1, "lines"),
    strip.text = element_markdown(face = "bold", size = 12),
    legend.margin = margin(l = -20, t = -20),
    legend.text = element_markdown(size = 12),
    legend.title = element_blank(),
    legend.position = "bottom",
  )

coverm_genus
plotly_build(coverm_genus)
htmlwidgets::saveWidget(as_widget(plotly_build(coverm_genus)), "graphs/plotly/genus.html")

```

```{r species_compare}

kmer_species_proportions <- species_proportions 
colnames(kmer_species_proportions) <- c("Species", "Assembler", "SampleName", "Relative Abundance") 
kmer_species_proportions <- kmer_species_proportions %>%
  add_column(Approach = c("K-mers"))

all_species_proportions <- coverm_gtdb[, c("Species", "Assembler", "Relative Abundance", "SampleName")] %>% 
  filter(str_trim(Species) != "") %>%
  aggregate(. ~ Species + Assembler + SampleName, FUN = sum) %>% 
  add_column(Approach = c("MAGs")) %>% 
  rbind(kmer_species_proportions)

coverm_species <- ggplot(all_species_proportions, aes(x = Assembler, y = `Relative Abundance`, fill = `Species`)) + 
  geom_bar(stat = "identity", position = "fill") +
  facet_wrap(~ SampleName) + 
  coord_flip() +
  xlab("") +
  ylab("") +
  #ggtitle("<b>Relative Abundance of MAGs, by Species</b>") +
  labs(fill = "Species") +
  scale_y_continuous(labels = scales::percent, breaks=c(0, 1)) +
  theme_classic() +
  theme(
    axis.text.y = element_markdown(face="bold", size = 14),
    axis.text.x = element_markdown(size = 12),
    plot.title = element_markdown(hjust = 0.5),
    panel.spacing = unit(1, "lines"),
    strip.text = element_markdown(face = "bold", size = 12),
    legend.margin = margin(l = -20, t = -20),
    legend.text = element_markdown(size = 12),
    legend.title = element_blank(),
    legend.position = "bottom",
  )

coverm_species
#plotly_build(coverm_species)
htmlwidgets::saveWidget(as_widget(plotly_build(coverm_species)), "graphs/plotly/species.html")

```
